ouroboros-network-api/src/Ouroboros/Network/SizeInBytes.hs

```diff
@@ -1,5 +1,15 @@
-module Ouroboros.Network.SizeInBytes (SizeInBytes) where
+{-# LANGUAGE DerivingVia #-}
+
+module Ouroboros.Network.SizeInBytes (SizeInBytes (..)) where
 
 import           Data.Word (Word32)
 
-type SizeInBytes = Word32
+import           NoThunks.Class (NoThunks (..))
+
+newtype SizeInBytes = SizeInBytes { getSizeInBytes :: Word32 }
+  deriving (Show, Eq, Ord)
+  deriving Enum     via Word32
+  deriving Num      via Word32
+  deriving Real     via Word32
```

Comment:
Do we need this instance? It seems strange that `SizeInBytes` is an instance of `Real`.

Reply:
`Real` just gives access to `toRational :: SizeInBytes -> Rational` which is useful when one needs to do some division.